backend/controllers/addGroup.test.js

```diff
@@ -0,0 +1,86 @@
+const { selectGroupFactory } = require('./addGroup')
+
+describe('select groups factory', () => {
+  const getUserAndGroups = jest.fn()
+  const saveGroup = jest.fn()
+  const logError = jest.fn()
+  const addGroup = selectGroupFactory(
+    getUserAndGroups,
+    saveGroup,
+    '/maintain-auth-users',
+    'Maintain auth users',
+    logError
+  )
+
+  describe('index', () => {
+    it('should call addGroup render', async () => {
+      const req = { params: { username: 'joe' }, flash: jest.fn() }
+      getUserAndGroups.mockResolvedValue([
+        [{ groupName: 'name', groupCode: 'code' }],
+        { username: 'BOB', firstName: 'Billy', lastName: 'Bob' },
+      ])
+
+      const render = jest.fn()
+      await addGroup.index(req, { render })
+      expect(render).toBeCalledWith('addGroup.njk', {
+        errors: undefined,
+        maintainTitle: 'Maintain auth users',
+        maintainUrl: '/maintain-auth-users',
+        groupDropdownValues: [{ text: 'name', value: 'code' }],
+        staff: { name: 'Billy Bob', username: 'BOB' },
+        staffUrl: '/maintain-auth-users/joe',
+      })
+    })
+
+    it('should copy any flash errors over', async () => {
+      const req = { params: { username: 'joe' }, flash: jest.fn().mockReturnValue({ error: 'some error' }) }
+      getUserAndGroups.mockResolvedValue([[], { username: 'BOB', firstName: 'Billy', lastName: 'Bob' }])
+
+      const render = jest.fn()
+      await addGroup.index(req, { render })
+      expect(render).toBeCalledWith('addGroup.njk', {
+        errors: { error: 'some error' },
+        maintainTitle: 'Maintain auth users',
+        maintainUrl: '/maintain-auth-users',
+        groupDropdownValues: [],
+        staff: { name: 'Billy Bob', username: 'BOB' },
+        staffUrl: '/maintain-auth-users/joe',
+      })
+    })
+
+    it('should call error on failure', async () => {
```

Comment:
Is it worth testing that logError is called correctly either here in another test?

Reply:
yes, will add in on next PR